### ISSUE TITLE
Hard-coded 'target/ift-repo' directory

### ISSUE THREAD

**Issue by azakrytnoi** [khmarbaise/maven-it-extension#171]
**Describe the bug**
Because this directory doesn't exists, test fails with the message:
`
[ERROR] nl.bkwi.twi.maven.JarExtractorTest.test_plugin_loaded(MavenExecutionResult)  Time elapsed: 0.058 s  <<< ERROR!
java.io.FileNotFoundException: Source '..../workspace/study/jar-extractor/target/itf-repo' does not exist
`
**To Reproduce**
`mvn clean verify` fails 100% repeatable.

`mkdir target/itf-repo
men verify
`
All fine.

**Current behavior**
Test fails because hard-hardcoded directory doesn't exists

**Expected behavior**
All necessary directories under `target` created automatically.

**Screenshots**

**The used versions**
The version of tools you have used.

 - Maven: 3.6.3]
 - JDK: OpenJDK 11.0.8]
 - IDE: IDEA IntelliJ 2020.2

**Additional context**

**Comment by khmarbaise** [comment]
Have you started from within IDE or from plain command line? Also helpful would be a full working example which can be used to reproduce this issue...

**Comment by khmarbaise** [comment]
Another which comes to my mind... The test you have created `nl.bkwi.twi.maven.JarExtractorTest` looks like a usual unit test and not like an integration test? Have you configured surefire plugin or failsafe plugin [accordingly to the documentation](https://khmarbaise.github.io/maven-it-extension/itf-documentation/usersguide/usersguide.html#_configuration_in_maven)?

**Comment by azakrytnoi** [comment]
Yes, all environment was created accordingly to documentation. By the way, tests in project fails exactly the same way. So full working examples: your own tests.

**Comment by khmarbaise** [comment]
Unfortunately this is not helpful because all my own tests and several other projects which are using this framework work fine without any issue... Please make a full working example which shows the issue ... otherwise I can't reproduce the problem...

**Comment by khmarbaise** [comment]
@azakrytnoi can you give some feedback here? Or is not relevant anymore?

**Comment by arunkjn** [comment]
Hi, I am also getting the same issue. I tried to follow the documentation and running this test through intellij

**Comment by arunkjn** [comment]
I found that the issue was happening because I did not run `maven install` after adding dependencies.

**Comment by khmarbaise** [comment]
An `mvn install` should be unrelated. That's the reason why I need an example project  which shows the problem.

**Comment by arunkjn** [comment]
After running `mvn clean install` the required folder `target/itf-repo` is generated and the problem goes away. In my case what happened is that I added the required dependencies to pom and generated the required structure for test file and resources as described in the docs. Now I was trying to run the integration test directly via my IDE intellij. I think in doing so, intellij does not install the required dependencies and tries to execute the test directly. Once I manually completed `mvn clean install` on the command line, the problem went away. 

I think it is safe to ignore this issue for now, as it seems to happen due to misconfiguration on user end.

**Comment by khmarbaise** [comment]
Please make a separate issue. I will explain it there in detail.

**Comment by khmarbaise** [comment]
Unfortunately neither getting feedback nor can I reproduce this problem because I don't have an example project.

**Comment by tadayosi** [comment]
I hit a similar issue but for me it was because it ran the integration-test phase before the itf-maven-plugin install goal was executed for some reason (for example, failsafe-maven-plugin is assigned to the test phase in the parent pom, etc). Sorting the execution order solved the issue for me.

**Comment by tadayosi** [comment]
@khmarbaise The execution order of your itf-maven-plugin. If it's somehow tied to the `test` phase this issue should occur. Not sure how doc can be improved but maybe you could note somewhere that the itf-maven-plugin has to run in the `integration-test` phase.

**Comment by JWT007** [comment]
Old issue but I just started using this and had the same problem.

After some playing around I figured out where I went wrong.
The setup documentation is a bit unclear about where the plugin definition for the 'itf-maven-plugin' belongs:

I had accidentally dropped it into my plugin-management so it wasn't running and the directories were not getting setup before the integration-test was running.  Once I moved it to 'build/plugins' the error was gone.

Maybe add the surrounding <project><build><plugins> to the documentation for clarity?
